File: src/KeywordIdentifier.js
// KeywordIdentifier.js
import React, { useState } from 'react';

const KeywordIdentifier = () => {
  const [text, setText] = useState('');
  const [keywords, setKeywords] = useState('');
  const [result, setResult] = useState('');

  const identificarPalavrasChave = () => {
    fetch('http://localhost:7000/identify', {
      method: 'POST',
      headers: {
        'Content-Type': 'application/json',
      },
      body: JSON.stringify({text}),
    })
    .then(response => response.json())
    .then(data => {
      setResult(data.keywords.join(', '));
    })
    .catch(error => {
      console.error('Erro ao identificar palavras-chave:', error);
    });
  };

  return (
    <div>
      <h2>Identificador de Palavras-Chave</h2>
      <label htmlFor="texto">Insira seu texto:</label>
      <textarea id="texto" rows="4" cols="50" value={text} onChange={e => setText(e.target.value)}></textarea>
      <br />
      <label htmlFor="palavrasChave">Palavras-Chave (separadas por vírgulas):</label>
      // <input type="text" id="palavrasChave" value={keywords} onChange={e => setKeywords(e.target.value)} />
      // <br />
      // <button onClick={identificarPalavrasChave}>Identificar Palavras-Chave</button>
      <p>Resultado: {result}</p>
    </div>
  );
};

export default KeywordIdentifier;

File: flu/lib/keyword_identifier.dart
import 'package:flutter/material.dart';
import 'package:http/http.dart' as http;




class KeywordIdentifierApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Keyword Identifier',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: KeywordIdentifierPage(),
    );
  }
}

class KeywordIdentifierPage extends StatelessWidget {
  final TextEditingController textController = TextEditingController();
  final TextEditingController keywordsController = TextEditingController();
  String result = '';

  Future<void> identifyKeywords() async {
    try {
      var response = await http.post(
        Uri.parse('http://localhost:4000/identify'),
        headers: {'Content-Type': 'application/json'},
        body: {'text': textController.text, 'keywords': keywordsController.text.split(',')},
      );

      var data = response.body;
      result = data;
    } catch (error) {
      print('Erro ao identificar palavras-chave: $error');
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: Text('Keyword Identifier'),
      ),
      body: Padding(
        padding: const EdgeInsets.all(16.0),
        child: Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: [
            Text('Identificador de Palavras-Chave'),
            TextField(
              maxLines: 4,
              minLines: 1,
              controller: textController,
              decoration: InputDecoration(labelText: 'Insira seu texto:'),
            ),
            SizedBox(height: 16),
            TextField(
              controller: keywordsController,
              decoration: InputDecoration(labelText: 'Palavras-Chave (separadas por vírgulas):'),
            ),
            SizedBox(height: 16),
            ElevatedButton(
              onPressed: () => identifyKeywords(),
              child: Text('Identificar Palavras-Chave'),
            ),
            SizedBox(height: 16),
            Text('Resultado: $result'),
          ],
        ),
      ),
    );
  }
}
